search/facets_builder.go

```diff
@@ -133,6 +133,17 @@ func (tf *TermFacets) Terms() []*TermFacet {
 	return tf.termFacets
 }
 
+func (tf *TermFacets) AvailableTerms() []*TermFacet {
```

Comment:
Would you add a supporting unit test for this please.

Reply:
hi @abhinavdangeti added a test

Reply:
@scottlepp Could you accomplish what you need with the existing `Terms()` func on `TermFacets`?

Reply:
The `Terms()` func returns `termFacets`, which is empty.  However you can see the `termsLookup` contains the terms.

![image](https://github.com/user-attachments/assets/6dba7e48-9599-49fe-8a98-eea006989a80)

Reply:
Just remembered that there are some other methods that change `termFacets`. `Fixup()` and `TrimToTopN()` can both remove (or entirely truncate) the `termFacets` slice.

I'm not that familiar with how these functions are called but, if I had to guess, it might only happen when the size of the facet request is very small (or zero).

```
bleve.NewFacetRequest("foo", 0)
```

As an aside, there doesn't appear to be a validation on the size of the facet request so perhaps there's a good reason to allow zero (I can't think of one, but again I'm only passingly familiar with the bleve code base).

Reply:
@tylerkovacs you're right.  the limit was defaulting to 0.